pkg/agent/proxy/proxier.go

```diff
@@ -264,6 +275,34 @@ func (p *proxier) installServices() {
 			}
 			endpointUpdateList = append(endpointUpdateList, endpoint)
 		}
+
+		// If the length of endpointUpdateList > maxEndpoints, endpointUpdateList should be cut. However, the iteration
+		// of map in Golang is random, so endpointUpdateList are not always in the same order. If endpointUpdateList is
+		// cut directly without any sorting, some Endpoints may not be installed on the cut endpointUpdateList(Last sync,
+		// these Endpoints may be cut off from endpointUpdateList, they are of course not installed.) So cutting
+		// endpointUpdateList after sorting can avoid this situation in some degree.
```

Comment:
I think it hasn't addressed the concern: https://github.com/vmware-tanzu/antrea/pull/2101#discussion_r619739242. The extra endpoints won't be in "endpointsInstalled" and "needUpdateEndpoints" will always be true regardless of how many times it has been reconciled, unless I misunderstand the code. Could you test it with one oversized service and one normal size service and check whether updating the latter one will cause the first one resynced?

Reply:
I'll try to add some test code to verify this.

Reply:
I don't mean to add test code to verify it as it will have to check logs and really create a service with 800+ endpoints. Could you just change maxEndpoints to 10 and verify it manually if you haven't done it before? From the code, I think it has the issue.

Reply:
OK, I'll verify this manually.

Reply:
any update?

Reply:
I have changed `maxEndpoints` to 4, then built and run. The detail test is below:

-  Applies a deployment of 6 pods and applies a Service selecting these 6 pods. E.g, this Endpoint IPs are (10.10.1.1-6)
- Endpoints whose IPs are 10.10.1.1-4 are selected(According to sort function of Endpoints, 10.10.1.1 will be in the first place after sorting, and 10.10.1.6 will be in the last place after sorting.). Related warning log appears and Service name will be recorded.
- When deleting the pod whose IP is 10.10.1.5-6, Service Group will not be updated as they are extra Endpoints and the Endpoints from new creating pods(assuming these two IPs are 10.10.1.7, 10.10.1.8) are also extra after sorting.  Related warning log doesn't appears as Service name is recorded.
- When deleting the pod whose IP is 10.10.1.1-4, Service Group will be updated. Related warning log doesn't appears as Service name is recorded.